motion_computation/src/motion_computation_worker.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "motion_computation_worker.h"
+#include <motion_predict/motion_predict.h>
+#include <motion_predict/predict_ctrv.h>
+#include <ros/ros.h>
+#include <cav_msgs/ExternalObject.h>
+#include <cav_msgs/ExternalObjectList.h>
+#include <autoware_msgs/DetectedObject.h>
+#include <autoware_msgs/DetectedObjectArray.h>  
+#include <functional>
+
+
+namespace object
+{
+MotionComputationWorker::MotionComputationWorker(const PublishObjectCallback& obj_pub) : obj_pub_(obj_pub){};
+
+void MotionComputationWorker::motionPredictionCallback(const cav_msgs::ExternalObjectList obj_list)//Duplicate input message
+{
+
+    cav_msgs::ExternalObjectList msg;
+
+    msg = predictionLogic(obj_list);
+ 
+  
+   obj_pub_(msg);
+
+}
+
+cav_msgs::ExternalObjectList MotionComputationWorker::predictionLogic(cav_msgs::ExternalObjectList obj_list)
```

Comment:
Based on the change to the callback you should modify this to look like this and modify the message in place. 
void predictionLogic(cav_msgs::ExternalObjectListPtr obj_list)